src/FirestoreProvider.js

```diff
@@ -32,13 +29,13 @@ export default class FirestoreProvider extends Component {
     };
   }
 
-  getChildContext() {
+  render() {
     const { firestoreDatabase, firestoreCache } = this.state;
 
-    return { firestoreDatabase, firestoreCache };
-  }
-
-  render() {
-    return this.props.children;
+    return (
+      <FirestoreContext.Provider value={{ firestoreDatabase, firestoreCache }}>
```

Comment:
@peternoordijk - We should change this to `value={this.state}`, so that we don't cause a re-render of all consumers whenever this component updates.

Here is some clarification on this from the react docs: https://reactjs.org/docs/context.html#caveats

Reply:
Good point! I will make a commit later today with this change.